decode.go

```diff
@@ -83,10 +83,10 @@ func (s *source) getFilepos() int64 {
 	return s.pos
 }
 
-// A Decoded is a MP3-decoded stream.
+// A Decoder is a MP3-decoded stream.
 //
-// Decoded decodes its underlying source on the fly.
-type Decoded struct {
+// Decoder decodes its underlying source on the fly.
+type Decoder struct {
```

Comment:
Because the reader in the struct does not decode all data in advance - Decoded would imply that the data is decoded immediately, which is not the case. Calling it a decoder is the idiomatically correct name, and is in line with how the stdlib and other libraries name objects that decode/deserialize/decompress things.